crypto/core_namemap.c

```diff
@@ -91,14 +92,7 @@ static const OPENSSL_CTX_METHOD stored_namemap_method = {
 
 int ossl_namemap_empty(OSSL_NAMEMAP *namemap)
 {
-    int rv = 0;
-
-    CRYPTO_THREAD_read_lock(namemap->lock);
-    if (namemap->max_number == 0)
-        rv = 1;
-    CRYPTO_THREAD_unlock(namemap->lock);
-
-    return rv;
+    return namemap == NULL || tsan_load(&namemap->max_number) == 0;
```

Comment:
What happens on platforms that don't have tsan support, i.e. where we access `namemap->max_number` unlocked?

Considering the urgency, I won't let this question block the PR, but we will eventually have to answer that question

Reply:
Bear in mind that the only check is if _max_number_ is zero and the only other operation on it is increment.  I'm not sure this is a problem unlocked but it could be replaced by a single byte flag to indicate if it's ever been incremented.  The increment is not done under an obvious lock.

Still, best to drop this commit and figure it out later.

Reply:
Another way is to make this function static.  It's used in only one spot in this same file...

Reply:
I thought about static, it means changes to the documentation (which are harmless).
Regardless, I've removed this commit for the moment -- this lock inversion is a symptom rather than the problem.